### PR TITLE
Use mvn sampler

### LINKED ISSUE #35: Use QuantEcon.MVNSampler in covariance_game
* [ ] Replace `Distributions.MvNormal` with `QuantEcon.MVNSampler` in `covariance_game`
* [ ] Add `rng::AbstractRNG` option to `random_game` and `covariance_game`

Refer to QuantEcon/QuantEcon.jl#157

### PR DESCRIPTION
**Pull request by QBatista**
Replaced MvNormal with MVNSampler in random.jl, added an option to choose RNG, and added a test for the corner values of rho.

Close #35

### CODE REVIEW

File: src/random.jl [hunk 1]
```diff
@@ -64,25 +70,29 @@ multi-normal with the covariance of any pair of payoffs equal to
...
 
-    if !(-1 / (N - 1) < rho < 1)
+    if !(-1 / (N - 1) <= rho <= 1)
         lb = (N == 2) ? "-1" : "-1/$(N-1)"
-        throw(ArgumentError("rho must be in ($lb, 1)"))
+        throw(ArgumentError("rho must be in [$lb, 1]"))
     end
 
     mu = zeros(N)
     C = fill(rho, (N, N))
```
Comment (shizejin) [inline src/random.jl]: The notation `C` follows `Distributions.jl`. Should we change it to `Sigma` to keep consistent with [MVNSampler](https://github.com/QuantEcon/QuantEcon.jl/blob/20a7e621792cc68bf49712354d0218cd61a0ded0/src/sampler.jl#L9)?

**Comment by shizejin** [comment]
Thanks @QBatista for doing this! I think [this line](https://github.com/QuantEcon/Games.jl/blob/master/src/Games.jl#L7) should also be deleted as we get rid of the dependency on `Distributions.jl`.

File: test/test_random.jl [hunk 2]
```diff
@@ -9,11 +9,29 @@
...
+        g = covariance_game(nums_actions, rho)
+        for a in CartesianRange(nums_actions)
+            payoff_profile = g[a]
+            for i in 1:(N-1)
+                @test payoff_profile[i] ≈ payoff_profile[end]
+            end
+        end
+
+        rho = -1/(N-1)
+        for a in CartesianRange(nums_actions)
```
Comment (shizejin) [inline test/test_random.jl]: Should there be `g = covariance_game(nums_actions, rho)` above?

**Comment by oyamad** [comment]
@shizejin Thanks for the comments. I agree on all these.

[Tests failed](https://travis-ci.org/QuantEcon/Games.jl/jobs/262501391), because there is "no method matching `getindex(::Games.NormalFormGame{3,Float64}, ::CartesianIndex{3})`".

**Comment by QBatista** [comment]
@shizejin Thanks for the comments -- I've now updated the code.

**Comment by oyamad** [comment]
Will you rebase on `master` (i.e., rebase your commits by choosing the latest commit on `master` as the parent), where the error on `getindex` is fixed.

**Comment by oyamad** [comment]
Or you can merge the `master` branch into your branch.

File: test/test_random.jl [hunk 3]
```diff
@@ -9,11 +9,30 @@
...
+            end
+        end
+
+        rho = -1/(N-1)
+        g = covariance_game(nums_actions, rho)
+        for a in CartesianRange(nums_actions)
+            payoff_profile = g[a]
+            for i in 1:N
+                @test sum(payoff_profile[i]) ≈ 0 atol=1e-10
+            end
```
Comment (oyamad) [inline test/test_random.jl]: This has to be `@test sum(payoff_profile) ≈ 0 atol=1e-10`.

**Comment by oyamad** [comment]
Merging. Good job, thanks!
